android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/ResourceDataRulesExecutorTest.kt

```diff
@@ -218,8 +219,106 @@ class ResourceDataRulesExecutorTest : RobolectricTest() {
+
+      val relatedRepositoryResourceData = mutableMapOf<String, LinkedList<Resource>>()
+      val computedValuesMap: Map<String, List<Resource>> = emptyMap()
+
+      relatedRepositoryResourceData[ResourceType.Patient.name] =
+        LinkedList<Resource>().apply { add(patient) }
+
+      val listResourceDataStateMap = mutableStateMapOf<String, SnapshotStateList<ResourceData>>()
+
+      resourceDataRulesExecutor.processListResourceData(
+        listProperties = listProperties,
+        relatedResourcesMap = relatedRepositoryResourceData,
+        computedValuesMap = computedValuesMap,
+        listResourceDataStateMap = listResourceDataStateMap
+      )
+
+      val snapshotStateList = listResourceDataStateMap[listProperties.id]
+      val resourceData = snapshotStateList?.first()!!
+      Assert.assertEquals(0, resourceData.computedValuesMap?.size)
+    }
+  }
+
+  @Ignore("To get fhirPathExpression to return stuff in retriveRelatedResources list")
+  @Test
+  @kotlinx.coroutines.ExperimentalCoroutinesApi
+  fun `processListResourceData with relatedResources and fhirPathExpression populates computedValuesMap`() {
+    runTest {
+      val registerCard = RegisterCardConfig()
+      val viewType = ViewType.CARD
+      val patient = Faker.buildPatient()
+      val listResource =
+        ListResource(
+          "id",
+          resourceType = ResourceType.Patient,
+          conditionalFhirPathExpression = "Patient.active",
+          relatedResources =
+            listOf(
+              ListResource(
+                null,
+                resourceType = ResourceType.Patient,
+                fhirPathExpression = "Patient.active = true"
+              )
+            )
+        )
+      val resources = listOf(listResource)
+      val listProperties =
+        ListProperties(registerCard = registerCard, viewType = viewType, resources = resources)
+
+      val relatedRepositoryResourceData = mutableMapOf<String, LinkedList<Resource>>()
+      val computedValuesMap: Map<String, List<Resource>> = emptyMap()
+
+      relatedRepositoryResourceData[ResourceType.Patient.name] =
+        LinkedList<Resource>().apply { add(patient) }
+
+      val listResourceDataStateMap = mutableStateMapOf<String, SnapshotStateList<ResourceData>>()
+
+      resourceDataRulesExecutor.processListResourceData(
+        listProperties = listProperties,
+        relatedResourcesMap = relatedRepositoryResourceData,
+        computedValuesMap = computedValuesMap,
+        listResourceDataStateMap = listResourceDataStateMap
+      )
+
+      val snapshotStateList = listResourceDataStateMap[listProperties.id]
+      val resourceData = snapshotStateList?.first()!!
+      Assert.assertEquals(1, resourceData.computedValuesMap?.size)
+      Assert.assertEquals("[TBD expected value]", resourceData.computedValuesMap?.values?.first())
+      Assert.assertEquals("[TBD expected value]", resourceData.computedValuesMap?.keys?.first())
```

Comment:
test that the map is populated correctly on https://github.com/opensrp/fhircore/pull/2449/files#diff-7f1a9ad0d64284e061a792f6a05d941c502896d5a68c4b4e4910fcb1eea530baR127 and other stuff in the condition defined on 126